src/swupd/files.go

```diff
@@ -0,0 +1,155 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+)
+
+type ftype int
+type fmodifier int
+type fstatus int
+
+const (
+	TYPE_UNSET ftype = iota
+	FILE
+	DIRECTORY
+	LINK
+	MANIFEST
+)
+
+const (
+	MODIFIER_UNSET fmodifier = iota
+	CONFIG
+	STATE
+	BOOT
+)
+
+const (
+	STATUS_UNSET fstatus = iota
+	DELETED
+	GHOSTED
+)
+
+// File represents an entry in a manifest
+type File struct {
+	Name    string
+	Hash    string
+	Version uint32
+
+	// flags
+	Type     ftype
+	Status   fstatus
+	Modifier fmodifier
+	Rename   bool
+
+	// renames
+	RenameScore uint16
+	RenamePeer  *File
+
+	Info      os.FileInfo
+	DeltaPeer *File
+}
+
+// set file type based on flag
+// F: file
+// D: directory
+// L: link
+// M: manifest
+// .: none (may be deleted/ghosted)
+func setFileTypeFromFlag(flag byte, f *File) error {
+	switch flag {
+	case 'F':
+		f.Type = FILE
+	case 'D':
+		f.Type = DIRECTORY
+	case 'L':
+		f.Type = LINK
+	case 'M':
+		f.Type = MANIFEST
+	case '.':
+	default:
+		return fmt.Errorf("invalid file type flag: %v", flag)
+	}
+
+	return nil
+}
+
+// set status based on flag
+// d: deleted
+// g: ghosted
+// .: exists
+func setStatusFromFlag(flag byte, f *File) error {
+	switch flag {
+	case 'd':
+		f.Status = DELETED
+	case 'g':
+		f.Status = GHOSTED
+	case '.':
+	default:
+		return fmt.Errorf("invalid file status flag: %v", flag)
+	}
+
+	return nil
+}
+
+// set modifier from string
+// C: config
+// s: state
+// b: boot
+func setModifierFromFlag(flag byte, f *File) error {
+	switch flag {
+	case 'C':
+		f.Modifier = CONFIG
+	case 's':
+		f.Modifier = STATE
+	case 'b':
+		f.Modifier = BOOT
+	case '.':
+	default:
+		return fmt.Errorf("invalid file modifier flag: %v", flag)
+	}
+
+	return nil
+}
+
+// set rename flag from string
+// r: this is a rename-from or renamed-to file
+func setRenameFromFlag(flag byte, f *File) error {
+	switch flag {
+	case 'r':
+		f.Rename = true
+	case '.':
+		f.Rename = false
+	default:
+		return fmt.Errorf("invalid file rename flag: %v", flag)
+	}
+
+	return nil
+}
+
+// set file flags from string
+func (f *File) SetFlags(flags string) error {
```

Comment:
Golint will complain about this comment if the function stays exported, should be in the format:
// SetFlags ...<blah blah>...

Reply:
This probably shouldn't be exported anyways, but I'll fix up the comments.